### ISSUE TITLE
_get can not return null values

### ISSUE THREAD

**Issue by 0x0ACB** [godotengine/godot#86989]
### Tested versions

4.2

### System information

Windows 10

### Issue description

According to the documentation and also my own testing `_get` can not return properties whose value is `null` even if it is included in `_get_property_list`

> Should return the given property's value, or null if the property should be handled normally.

This causes an issue cause if the property does indeed have the value null, accessing it will result in an error:
> Invalid get index '<var>'
Doing `return null as Variant` also does not work!

### Steps to reproduce

Implement `_get_property_list` and `_get` in a script
Return `null` for some of the properties in the property list
Try to access said properties

### Minimal reproduction project (MRP)

[CustomProperty.zip](https://github.com/godotengine/godot/files/13871316/CustomProperty.zip)

**Comment by Mickeon** [comment]
Yeah this has been a problem for a long while. In reality, whether `get()` returns `null` or not is used to determine if the Object contains the property.

Because of this by the way, in GDScript `my_property in my_object` will return `false` if `my_property` is null, too.

I rewrote the documentation back then to try and go around this quirk, and I have casually suggested a `has_property()` to address this in the past, but sometimes it's just the way it is.

**Comment by AThousandShips** [comment]
The problem here is that we can't pass values through arguments, like `_get(property, out ret) -> bool` would solve this, and is how it works in c++ and extension, but not possible in GDScript

**Comment by 0x0ACB** [comment]
One option would to have a special return object either `PropertyDoesNotExist` or `NullVariant` that are not null but get converted by the code that calls `_get`. With `NullVariant` this would be without breaking anything as well.

**Comment by Mickeon** [comment]
This is asking for a difference between "No return value" (void) and "nothing" (null), which are currently one in the same in Godot. That'd be a lot to change, even internally. But if it really is necessary, perhaps the `has_property()` solution would be more doable.

**Comment by 0x0ACB** [comment]
My suggestion was not to differentiate this in the general case. But rather for `_get` i.e implement `_get` as either:

```
func _get(property):
 if not has_property(property):
  return PropertyDoesNotExist()
 return properties[property]
 ```
or
```
func _get(property):
 if not has_property(property):
  return null
 return properties[property] or NullVariant
```
